lib/utils/fs.go

```diff
@@ -144,3 +153,59 @@ func StatDir(path string) (os.FileInfo, error) {
 	}
 	return fi, nil
 }
+
+// FSTryWriteLock tries to grab write lock, returns ErrUnsuccessfulLockTry
+// if lock is already acquired by someone else
+func FSTryWriteLock(filePath string) (unlock func() error, err error) {
+	fileLock := flock.New(filePath + lockPostfix)
+	locked, err := fileLock.TryLock()
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	if !locked {
+		return nil, trace.Wrap(ErrUnsuccessfulLockTry)
+	}
+
+	return unlockWrapper(fileLock.Unlock, fileLock.Path()), nil
+}
+
+// FSTryWriteLockTimeout tries to grab write lock, it's doing it until locks is acquired, or timeout is expired,
+// or context is expired.
+func FSTryWriteLockTimeout(ctx context.Context, filePath string, timeout time.Duration) (unlock func() error, err error) {
+	fileLock := flock.New(filePath + lockPostfix)
+	timedCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	if _, err := fileLock.TryLockContext(timedCtx, 10*time.Millisecond); err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+
+	return unlockWrapper(fileLock.Unlock, fileLock.Path()), nil
+}
+
+// FSTryReadLock tries to grab write lock, returns ErrUnsuccessfulLockTry
+// if lock is already acquired by someone else
+func FSTryReadLock(filePath string) (unlock func() error, err error) {
+	fileLock := flock.New(filePath + lockPostfix)
+	locked, err := fileLock.TryRLock()
+	if err != nil {
+		return nil, trace.ConvertSystemError(err)
+	}
+	if !locked {
+		return nil, trace.Wrap(ErrUnsuccessfulLockTry)
+	}
+
+	return unlockWrapper(fileLock.Unlock, fileLock.Path()), nil
+}
+
+// FSTryReadLockTimeout tries to grab read lock, it's doing it until locks is acquired, or timeout is expired,
+// or context is expired.
+func FSTryReadLockTimeout(ctx context.Context, filePath string, timeout time.Duration) (unlock func() error, err error) {
+	fileLock := flock.New(filePath + lockPostfix)
```

Comment:
To simplify and avoid confusion, it's better if you wrap the lock file name in a function.
```
fileLock := flock.New(getPlatformLockFile(filePath))
```
I think this is more clear than filename+lockPostfix.
What do you think?

Reply:
Yep, might be good idea, will change it.